groupmne/utils.py

```diff
@@ -234,18 +234,23 @@ def _floyd_warshall(dist):
 
 def _compute_ground_metric(src, group_info):
     """Compute geodesic distance matrix on the triangulated mesh of src."""
-    vertnos_filtered = group_info["vertno_ref"]
+    common_order = group_info["common_order"]
     hemis = ["lh", "rh"]
     Ds_f, Ds = [], []
     for i, h in enumerate(hemis):
         tris = src[i]["use_tris"]
         vertno = src[i]["vertno"]
         points = src[i]["rr"][vertno]
 
-        vert_used = vertnos_filtered[i]
+        vert_used_idx = common_order[i]
+
+        # if vertno are not 0 to n_points, change the numbering
+        if np.max(tris) > len(np.unique(tris)):
+            for ii, v in enumerate(vertno):
+                tris[tris == v] = ii
```

Comment:
you should do it on a copy of tris

Reply:
yup, done

Reply:
I think it's still dangerous to do as is. if you have some v and match some ii from before it will break.

Reply:
I see what you mean but since vertno is sorted, we always have ii <= v. We are always replacing large ints with smaller ints,  so an element of tris cannot be overwritten twice. Perhaps we add an assert for sorting ?